tests/by-util/test_ls.rs

```diff
@@ -6311,3 +6311,332 @@ fn test_ls_time_sort_without_long() {
+        .succeeds()
+        .stdout_move_str();
+    assert!(
+        result2.contains("\x1b["),
+        "Explicit --color should work even with -f before"
+    );
+}
+
+#[test]
+fn test_f_overrides_big_a() {
+    // Test last-flag-wins between -f and -A using .. as discriminator
+    // -f shows .. (all files including . and ..)
+    // -A hides .. (almost all, excluding . and ..)
+    let scene = TestScenario::new(util_name!());
+    let at = &scene.fixtures;
+    at.touch("visible");
+    at.touch(".hidden");
+
+    // -A then -f: -f wins, should show ..
+    let out_a_f = scene
+        .ucmd()
+        .arg("-A")
+        .arg("-f")
+        .arg("-1")
+        .succeeds()
+        .stdout_move_str();
+    assert!(
+        out_a_f.lines().any(|l| l == ".."),
+        "-f should win and show .."
+    );
+
+    // -f then -A: -A wins, should NOT show ..
+    let out_f_a = scene
+        .ucmd()
+        .arg("-f")
+        .arg("-A")
+        .arg("-1")
+        .succeeds()
+        .stdout_move_str();
+    assert!(
+        !out_f_a.lines().any(|l| l == ".."),
+        "-A should win and hide .."
+    );
+}
+
+#[test]
+fn test_f_overrides_sort_flags() {
+    // Test last-flag-wins using size-based sorting for determinism
+    let scene = TestScenario::new(util_name!());
+    let at = &scene.fixtures;
+
+    // Create files with different sizes for predictable sort order
+    at.write("small.txt", "a"); // 1 byte
+    at.write("medium.txt", "bb"); // 2 bytes  
+    at.write("large.txt", "ccc"); // 3 bytes
+
+    // Get baseline outputs (include -a to match -f behavior which shows all files)
+    let out_s = scene
+        .ucmd()
+        .arg("-S")
+        .arg("-a")
+        .arg("-1")
+        .succeeds()
+        .stdout_move_str();
+    let out_u = scene
+        .ucmd()
+        .arg("-U")
+        .arg("-a")
+        .arg("-1")
+        .succeeds()
+        .stdout_move_str();
+
+    // -f then -S: -S wins, should be sorted by size
+    let out_f_s = scene
+        .ucmd()
+        .arg("-f")
+        .arg("-S")
+        .arg("-1")
+        .succeeds()
+        .stdout_move_str();
+    assert_eq!(
+        out_f_s, out_s,
+        "-S should win: output should be size-sorted"
+    );
+
+    // -S then -f: -f wins, should be unsorted
+    let out_s_f = scene
+        .ucmd()
+        .arg("-S")
+        .arg("-f")
+        .arg("-1")
+        .succeeds()
+        .stdout_move_str();
+    assert_eq!(
+        out_s_f, out_u,
+        "-f should win: output should match unsorted -U"
+    );
+}
+
+#[test]
+fn test_a_overrides_f_files() {
+    // Test that -a after -f still shows all files
+    let scene = TestScenario::new(util_name!());
+    let at = &scene.fixtures;
+    at.touch("visible");
+    at.touch(".hidden");
+
+    scene
+        .ucmd()
+        .arg("-f")
+        .arg("-a")
+        .succeeds()
+        .stdout_contains(".hidden")
+        .stdout_contains("visible");
+}
+
+#[test]
+fn test_big_u_overrides_f_sort() {
```

Comment:
I'm not sure whether you forgot to use `-f` in this test or the function name is incorrect.